specs/vep-1493-vault-integration/README.md

```diff
@@ -0,0 +1,223 @@
+
+# VEP-1493: Vault Integration For Secrets Storage
+
+* **Author(s):** Dako Dakov (ddakov@gmail.com), ...
+* **Status:** draft
+
+
+To get started with this template:
+-
+- [ ] **Fill out this file as best you can.**
+  There are instructions as HTML comments.
+  At minimum, you should fill in the "Summary" and "Motivation" sections.
+- [ ] **Create a PR for this VEP.**
+- [ ] **Merge early and iterate.**
+  Avoid getting hung up on specific details and instead aim to get the goals of
+  the VEP clarified and merged quickly. The best way to do this is to just
+  start with the high-level sections and fill out details incrementally in
+  subsequent PRs.
+
+<!-- Provide table of content as it's helpful. -->
+
+- [Summary](#summary)
+- [Glossary](#glossary)
+- [Motivation](#motivation)
+- [Requirements and goals](#requirements-and-goals)
+- [High-level design](#high-level-design)
+- [API Design](#api-design)
+- [Detailed design](#detailed-design)
+- [Implementation stories](#implementation-stories)
+- [Alternatives](#alternatives)
+
+## Summary
+
+Provide integration with Hashicorp Vault for secure storage of secrets. Control Service operators will be able to
+optionally configure a vault instance to be used for storing data job properties which are marked as secret.
```

Comment:
a. It seems the proposal is dual properties storage, where entries marked as secret go to another data store. Is that correct?

b. Is the feature meant to be reused by other plugins, swapping the secret storage to another one than Vault? Asking about properties conditional multi-storage routing, and reusability

c. Is there any reason for excluding the more basic use case for storing all properties to Vault?

Reply:
The a. and b. are referring the current design suggested.

I guess the separate PR is about c. swapping the entire backing storage that is simpler use case. I feel like connecting to Vault storage is an implementation step, preceding the conditional routing step.

Reply:
I've added a couple of points in the Alternatives section for clarity.